apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.html

```diff
@@ -1,290 +1,253 @@
+                <button
+                    mat-stroked-button
+                    color="primary"
+                    type="button"
+                    [matMenuTriggerFor]="entityTypeMenu"
+                >
+                    {{ selectedFilter?.entityType }} <span class="caret"></span>
+                </button>
+                <mat-menu #entityTypeMenu="matMenu">
+                    <ng-container *ngFor="let entity of gstr1entityType">
+                        <button mat-menu-item aria-label="download" (click)="viewFilteredTxn('entityType', entity?.value)">
+                            {{ entity.label }}
                         </button>
-                        <ul *dropdownMenu class="dropdown-menu" role="menu" aria-labelledby="button-basic">
-                            <li *ngFor="let entity of gstr1entityType">
-                                <a
-                                    class="dropdown-item"
-                                    href="javascript:void(0)"
-                                    (click)="viewFilteredTxn('entityType', entity?.value)"
-                                    >{{ entity.label }}</a
-                                >
-                            </li>
-                        </ul>
-                    </div>
-                    </div>
-                    <div class="filter-row" *ngIf="selectedGst === GstReport.Gstr2 && filterParam.type === 'b2bur'">
-                        <div class="name">{{ localeData?.filing?.transaction_type }}:</div>
-                    <div class="btn-group filter d-inline-block" dropdown>
-                        <button
-                            id="button-basic2"
-                            dropdownToggle
-                            type="button"
-                            class="btn btn-primary dropdown-toggle"
-                            aria-controls="dropdown-basic"
-                        >
-                            {{ selectedFilter?.entityType }} <span class="caret"></span>
+                    </ng-container>
+                </mat-menu>
+            </div>
+            </div>
+            <div class="filter-row" *ngIf="selectedGst === GstReport.Gstr2 && filterParam.type === 'b2bur'">
+                <div class="name">{{ localeData?.filing?.transaction_type }}:</div>
+            <div class="btn-group filter d-inline-block">
+                <button
+                    mat-stroked-button
+                    color="primary"
+                    type="button"
+                    [matMenuTriggerFor]="entityTypeMenu"
+                >
+                    {{ selectedFilter?.entityType }} <span class="caret"></span>
+                </button>
+                <mat-menu #entityTypeMenu="matMenu">
+                    <ng-container *ngFor="let entity of gstr2entityType">
+                        <button mat-menu-item aria-label="download" (click)="viewFilteredTxn('entityType', entity?.value)">
+                            {{ entity.label }}
```

Comment:
```
```suggestion
                            {{ entity?.label }}
```

Reply:
no need it will not throw error at any condition